packages/react/src/components/Provider/Provider.tsx

```diff
@@ -152,9 +171,22 @@ class Provider extends React.Component<ProviderProps> {
           if (isBrowser()) render(outgoingTheme.renderer)
           this.renderStaticStylesOnce(outgoingTheme)
 
+          const rtlProps: { dir?: 'rtl' | 'ltr' } = {}
+          // only add dir attribute for top level provider or when direction changes from parent to child
+          if (
+            !incomingTheme ||
+            (incomingTheme.rtl !== outgoingTheme.rtl && _.isBoolean(outgoingTheme.rtl))
+          ) {
+            rtlProps.dir = outgoingTheme.rtl ? 'rtl' : 'ltr'
+          }
+
           return (
             <RendererProvider renderer={outgoingTheme.renderer} {...{ rehydrate: false }}>
-              <ThemeProvider theme={outgoingTheme}>{children}</ThemeProvider>
+              <ThemeProvider theme={outgoingTheme}>
+                <ElementType {...unhandledProps} {...rtlProps} className={classes.root}>
```

Comment:
Let's make ElementType ProviderBox so that the user can't break the implementation.  Forward the as prop from the Provider to the ProviderBox.